### PR TITLE
Remove Conic option from Color Picker

### LINKED ISSUE #1433: Color Picker - Remove Conic Gradient
## Feature Description

Per this discussion on[ Slack with Sam](https://xwp.slack.com/archives/CCCAUAH6F/p1588279028158100?thread_ts=1588262974.144900&cid=CCCAUAH6F), we will temporarily remove the Conic Gradient

#1389 will be picked up at a later date

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria

1. The conic gradient is removed from the editor
2. Note if helpful for future work, save the conic gradient code as we may add it back later

### PR DESCRIPTION
**Pull request by miina**
## Summary
Removes 'conic' option for color picker and adds migration for existing elements.

## To-do

- [x] Should we migrate style presets already? ->> Looks like we do, this PR is blocked by #754

---

Fixes #1433

### CODE REVIEW

**Comment by miina** [comment]
@dvoytenko Do you think it's important to create migration logic for style presets already now, considering it wasn't part of alpha?

In case of style presets, the migration would probably remove the style presets which use `conic`, however, not sure if it's relevant currently. Thoughts?

**Comment by swissspidy** [comment]
@miina Migrating style presets would require #754, correct?

**Comment by miina** [comment]
> Migrating style presets would require #754, correct?

Sounds like it, however, it could potentially just filter it dynamically without making actual changes in the DB -- just for passing the expected data to the app. The editor would then update the data when saving the story.

That's why I was wondering about the necessity for style presets migration -- since that would require a bit more changes that perhaps weren't originally considered in the scope of this issue. :)
Currently, the editor throws an error if someone has previously saved a preset with `conic`.

Happy to look into #754 though within this PR if we're going to need it anyway!

**Comment by dvoytenko** [comment]
> @dvoytenko Do you think it's important to create migration logic for style presets already now, considering it wasn't part of alpha?

Document-wise, I think we don't have to. But, indeed, not sure what to do with DB.

**Comment by swissspidy** [comment]
@miina The DB upgrade part should now be unblocked as #1751 was just merged

File: assets/src/edit-story/migration/migrations/v0018_conicToLinear.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+function updateElement(element) {
+  const { backgroundColor } = element;
+  if (backgroundColor?.type !== 'conic') {
+    return element;
+  }
+  return {
+    ...element,
+    backgroundColor: {
+      ...backgroundColor,
+      type: 'linear',
```
Comment (barklund) [inline assets/src/edit-story/migration/migrations/v0018_conicToLinear.js]: Conic gradients can have a `center`, linear gradients can't. However, there's currently no way to manipulate the center, so no conic gradients will have one. So this is just a convoluted way of saying, that we're good 🤝

File: assets/src/edit-story/utils/generatePatternStyles.js [hunk 2]
```diff
@@ -94,7 +82,7 @@ function getGradientDescription({ type, rotation, center, size }) {
  *
  * @return {Array} List of serialized stops
  */
-function getStopList(stops, alpha, isAngular) {
+function getStopList(stops, alpha, isAngular = false) {
```
Comment (barklund) [inline assets/src/edit-story/utils/generatePatternStyles.js]: `isAngular` is not needed now, as all both linear and gradient stops can be described using `%`. This parameter can be removed and logic below can be collapsed.

**Review by barklund — COMMENTED** [review]
Just a single comment about unnecessary code - otherwise looks good.

File: assets/src/edit-story/utils/generatePatternStyles.js (hunk 2 above)
Comment (miina) [reply assets/src/edit-story/utils/generatePatternStyles.js]: Removed in 9a9fb4e

**Review by barklund — APPROVED** [review]
Looks good to me!

**Comment by miina** [comment]
@swissspidy / @spacedmonkey Could you review the PHP migration part of the PR?

File: includes/Database_Upgrader.php [hunk 3]
```diff
@@ -56,9 +58,11 @@ public function init() {
 		$version  = get_option( self::OPTION, '0.0.0' );
 		$routines = [
 			'1.0.0' => 'upgrade_1',
+			'2.0.0' => 'v_2_replace_conic_style_presets',
 		];
 
 		array_walk( $routines, [ $this, 'run_upgrade_routine' ], $version );
+		// @todo This should only update if there were actual updates. Otherwise, the `self::PREVIOUS_OPTION` will always be overwritten.
```
Comment (swissspidy) [inline includes/Database_Upgrader.php:65]: @spacedmonkey Can you take a look at this in a new issue / PR?

**Review by swissspidy — APPROVED** [review]
LGTM on the PHP part

**Pull request merged**